lib/src/breakpoints_utils.dart

```diff
@@ -1,40 +1,62 @@
-import 'package:adaptive_breakpoints/adaptive_breakpoints.dart';
-
 import 'package:flutter/material.dart';
+import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';
 
-extension Breakpoints on BuildContext {
-  /// All the breakpoints according to Material Design's breakpoint system.
-  BreakpointSystemEntry get breakpoints => getBreakpointEntry(this);
-
-  /// Double the default gutter size.
-  double get gutterLarge => (_material3Value ?? breakpoints.gutter) * 2;
-
-  /// The gutter size according to Material Design's breakpoints system.
-  double get gutter => _material3Value ?? breakpoints.gutter;
-
-  /// Half the default gutter size.
-  double get gutterSmall => (_material3Value ?? breakpoints.gutter) / 2;
-
-  /// A quarter the default gutter size.
-  double get gutterTiny => (_material3Value ?? breakpoints.gutter) / 4;
+import '../flutter_gutter.dart';
 
+/// A gap of the standard gutter size according to Material Design's breakpoints
+extension BreakpointExtensions on BuildContext {
   /// The margin size according to Material Design's breakpoints system.
-  double get margin => _material3Value ?? breakpoints.margin;
-
-  int get columns => breakpoints.columns;
+  double get margin => materialSpacing;
+
+  /// The size according to Material Design's with a specified [GutterType].
+  double gutter({GutterType type = GutterType.medium, double scaleFactor = 1}) {
+    return scaleFactor *
+        switch (type) {
+          GutterType.tiny => materialSpacing / Gutter.scaleFactorMediumDefault,
+          GutterType.small => materialSpacing / Gutter.scaleFactorSmallDefault,
+          GutterType.medium => materialSpacing,
+          GutterType.large => materialSpacing * Gutter.scaleFactorSmallDefault,
+          GutterType.extraLarge =>
+            materialSpacing * Gutter.scaleFactorMediumDefault,
+          GutterType.expand => double.infinity,
+        }
+      ..floor();
```

Comment:
Why are we adding the `floor` here? Shouldn't Flutter handle any non-pixel boundaries when it converts to physical pixels?

Also, if we DO want to keep the floor in, shouldn't it apply after multiplying by scale factor?

Either way I'll just merge and ship this now and if we want to pull the floor back out I'll handle that in a patch version because I don't want to leave you stuck in async review h*ll.

Reply:
I'm making some more changes right now. Just found some stuff. I'll also add some tests